yt_idefix/data_structures.py

```diff
@@ -658,7 +658,20 @@ def _parse_definitions_header(self) -> None:
 
     def _get_input_parameter(self, match: re.Match) -> str:
         """Replace matched input parameters with its value"""
+        if not self._inifile:
+            warnings.warn(
+                "The inifile is missing for unit definitions. "
+                f"Specify it to the keyword argument 'inifile' (default: {self._default_inifile}), "
+                "or make sure that all fields are in code units!"
+            )
+            return "1.0"
         key = match.group(1)
+        if key not in self.parameters["Parameters"]:
+            warnings.warn(
+                f"Cannot get the value of {key} from {self._inifile}."
```

Comment:
this is the only time we actually require `self._inifile` in this function, so it would make more sense to check if it exists _now_ and adapt the warning message as needed

Reply:
I just considered a case where there is only an incorrect inifile in the directory, maybe the `pluto.ini` generated by Pluto's setup script. In this case, although the inifile exists, the value of the specific parameter still cannot be returned and it will raise an error.

It might be a very rare case. I just want to make it clear where the error was raised. If you think it's unnecessary, I can drop it and the corresponding test.

Reply:
@neutrinoceros I guess you might miss this message :) Could you give me your opinion on my consideration? Then I will modify it according to your requests.

A brief summary to help you recall the work in this PR: I designed two warning messages for the cases where inifile is missing and an incorrect inifile is given (without required parameters inside), and also two corresponding tests.

Reply:
Thanks for reminding me, indeed it seems I missed your first reply !
I'm not arguing *against* the new warning you're adding. What I meant is that the earlier check (currently at line 661) should be moved here, because this function shouldn't *rely on* `self._inifile`: it's just convenient to include it in the current warning, but really we don't need it. Is that any clearer ?

Reply:
Sure, I get your point now!